plugins/core/gs-appstream.c

```diff
@@ -771,6 +771,11 @@ gs_appstream_refine_app (GsPlugin *plugin,
 	if (as_app_has_category (item, "featured"))
 		gs_app_add_kudo (app, GS_APP_KUDO_FEATURED_RECOMMENDED);
 
+	/* Mark com.endlessm. apps with featured kudo. See discussion on T23152 */
+	if (!gs_app_has_kudo (app, GS_APP_KUDO_FEATURED_RECOMMENDED) &&
+	     g_str_has_prefix (gs_app_get_id (app), "com.endlessm."))
+		gs_app_add_kudo (app, GS_APP_KUDO_FEATURED_RECOMMENDED);
```

Comment:
If it already has the `GS_APP_KUDO_FEATURED_RECOMMENDED`, then what's the point of adding it again? This is a no-op. What's setting `GS_APP_KUDO_FEATURED_RECOMMENDED` in the first place?